### DISCUSSION TITLE
Camera with different model names

### DISCUSSION THREAD

**Discussion by artincube** [photoprism/photoprism#3565] [Q&A]
I have mainly this with A7r. I think thethered shots and such have different model names depending on the software used. I'd like them to be all same as if out of the camera. To batch this change, can I use exitool on the originals and photoprism will get it at next index or there is maybe a way in photoprism (not one by one)?

**Comment by lastzero** [comment]
We have a lookup table for this somewhere in our repository. I'm on my phone now, so can't browse the code. GigHub has a search function though eg search for Kodak. You could probably also update the cameras table in the database if you know a little SQL.

**Reply by artincube** [reply]
argh no SQL no. so for my question, if exiftool updates the tags, photoprism will reindex the files and get their new names?

**Reply by lastzero** [reply]
Yes, but then you need to do this every time. If we add the normalization to PhotoPrism, it will happen automatically without needing to update any files.

**Reply by artincube** [reply]
got it. I am not shooting tethered any more (or should I say), so probably once for me. Thanks. Thanks for your work and dedication.

**Comment by artincube** [comment]
Back to this for a clarification: Actually some camera model names are not anywhere else than in Photoprism:
264 RX-100
264  ILCE-7R

I guess 264 is for H.264 as they are all videos. Correct?
Photoprism does differentiate camera model name depending on still/video correct?
Something not to be found in exiftool, digikam or any metadata.

My opinion is that keeping the original camera name is better; it is easy enough to find only videos already.

**Reply by lastzero** [reply]
The normalization tables for makes and models that I referred to can be found here:

- https://github.com/photoprism/photoprism/blob/develop/internal/entity/camera_makes.go

However, I can't find "264 RX-100" or "264 ILCE-7R" anywhere in our code? I also don't know what 264 means in this context. PhotoPrism will use / show whatever is in the metadata of the files you index. If it's a stack, then the primary file is used for this.

**Reply by artincube** [reply]
I checked that list.
May I attach some screenshots of some of camera list in the webgui.
I tried to look at the tags of some files with exiftool:
-no references to these names for Sony, not a tag with 264.
-Yes for Nikon and Insta360 (Photoprism got the names in their metadata), So I can change these with exiftool on the originals the way I wish.

Sony: these "264" names have only videos shot with the respective cameras.
Nikon E5400 (Coolpix 5400): same, one camera name has all the videos and the other all the stills.
Samsung: 2 of these are Samsung S7 from different continents

So really only the Sony 264 names are out of my control, unless somehow I missed a tag (but I used -ee to extract tags).

You were talking about manipulating the camera table in the database....
That's my control, is it?


<img width="248" alt="ScreenShot 2023-07-21 at 9 26 28 AM" src="https://github.com/photoprism/photoprism/assets/68607302/f50939e0-4f1f-4e1c-99f6-7236858a944f">
<img width="243" alt="ScreenShot 2023-07-21 at 9 26 59 AM" src="https://github.com/photoprism/photoprism/assets/68607302/12b8cf34-7667-4e99-9d4b-acb9e17ca09e">
<img width="247" alt="ScreenShot 2023-07-21 at 9 27 19 AM" src="https://github.com/photoprism/photoprism/assets/68607302/f52b547c-7ed8-4093-8f91-3a7be06d137f">
<img width="235" alt="ScreenShot 2023-07-21 at 9 27 36 AM" src="https://github.com/photoprism/photoprism/assets/68607302/b065bce0-e0d8-48f2-8abd-e22649aeb04d">
<img width="254" alt="ScreenShot 2023-07-21 at 9 28 03 AM" src="https://github.com/photoprism/photoprism/assets/68607302/ba37f049-21b1-4489-a62f-1a150830c060">

**Reply by lastzero** [reply]
In an act of self-defense, we're going to take this weekend off. It may or may not be that the metadata you see has nothing to do with your files. However, I know that there is no code that just makes things up. Here is a complete list of supported tags:
- https://www.photoprism.app/kb/metadata

If you can provide us with a file and describe exactly what is wrong with the indexing, I'll be happy to take a look next week. Ideally, include a screenshot of the data that Exiftool displays when you run it, and also check if there are any related sidecar files from which metadata could be extracted.

**Reply by artincube** [reply]
I'll try do all this. 
Have all a good week end.
